core/src/main/resources/hudson/tasks/Fingerprinter/help.html

```diff
@@ -10,14 +10,14 @@
     <li>
       My BAR project depends on <tt>foo.jar</tt> from the FOO project.
     </li>
-    <li><ul>
+    <ul>
       <li>
         Which build of <tt>foo.jar</tt> is used in BAR #51?
       </li>
       <li>
         Which build of BAR contains my bug fix to <tt>foo.jar</tt> #32?
       </li>
-    </ul></li>
+    </ul>
```

Comment:
As per https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ul Every nested ul tag needs to be a child of a il element, so I propose going back to `<li><ul></ul></li>`

Reply:
So, this change was because the rendering was bad.

It's functionally:    
* I have <tt>foo.jar</tt> on my HDD but which build number of FOO did it come from?
* My BAR project depends on <tt>foo.jar</tt> from the FOO project.
*
  * Which build of <tt>foo.jar</tt> is used in BAR #51?
  * Which build of BAR contains my bug fix to <tt>foo.jar</tt> #32?

It's really unclear to me what the goal is.

Vaguely reasonable rendering one:
* I have <tt>foo.jar</tt> on my HDD but which build number of FOO did it come from?
* My BAR project depends on <tt>foo.jar</tt> from the FOO project.
  * Which build of <tt>foo.jar</tt> is used in BAR #51?
  * Which build of BAR contains my bug fix to <tt>foo.jar</tt> #32?

Vaguely reasonable rendering two:
* I have <tt>foo.jar</tt> on my HDD but which build number of FOO did it come from?
* My BAR project depends on <tt>foo.jar</tt> from the FOO project.
* Which build of <tt>foo.jar</tt> is used in BAR #51?
* Which build of BAR contains my bug fix to <tt>foo.jar</tt> #32?

The bigger question is: are the "which things" children of "my bar project depends"?

Reply:
Ah, so what you meant was to have a second level bullet list and the problem was that there was an extra bullet point like in your first example.

I think the issue comes from originally trying to set the indented `<ul>` in a separate `<li>`. To fix it, just set the indented list in the same `<li>` as the "title" of the list. I made a codepen with an example: https://codepen.io/nicupascu/pen/gEPmbb

Let me know if this was the issue or I misunderstood, thanks

Reply:
The question is whether that's correct. I didn't fix it that way because it wasn't apparent to me if the `<ul>` is really part of that item (`<li>`). It's still not apparent to me.

You're looking at this from a purely html perspective, you need to take that hat off and look at it from a language perspective.

In what world does the statement `My bar project...` somehow "encompass" `Which ...` and `Which ...`? And how/why does it encompass those two items more tightly than `I have...`.

If it doesn't, then that's the wrong answer.

Reply:
I see your point, but I do not agree. The new list is not part of the li, it is indented from that li, at least that is how I see it. This change would not break the markup, visually it looks ok but we would be introducing invalid html, which personally I don't agree with because future browsers might decide to render this case in strange ways like introducing a closing li and ul tag and open a brand new list. If other people don't mind this, I won't insist further.

Reply:
I spent years working in w3 on HTML and other standards. I'm not interested in speaking with a standards person on this point. HTML as a standard is designed to help offer semantic content. Here the problem is the semantics and only someone who understands that can help. As you aren't offering to help with that, I accept your offer to discontinue the discussion.

Reply:
>If it doesn't, then that's the wrong answer.

so based on the current rendering what is the correct answer - because the rendering which if option 2 is almost certainly it does encompass it.  so given you want the rendering like that you seem to have answered your onw question, and then Nicu's comment about the closing `</li>` being in the wrong place is correct.

reading them it is clear that "My BAR project depends on foo.jar from the FOO project." is a statement not attached to fingerprints and thus the following 2 bulltes belong to that as a sub statements.

thus option one is correct, and thus the html can be valid and the full stop `of My BAR project depends on <tt>foo.jar</tt> from the FOO project.` should be replaced by a colon.

Reply:
Agree with @jtnord, it's basically:

> This answers questions like:
> * Question 1?
> * Precondition for further questions
>   * Question 2 referencing precondition?
>   * Question 3 referencing precondition?

A reasonable alternative could be:

> With precondition for some questions, this answers questions like:
> * Question 1?
> * Question 2 referencing precondition?
> * Question 3 referencing precondition?

Reply:
To be clear, the only rendering preference I have is **NOT** this:

* I have <tt>foo.jar</tt> on my HDD but which build number of FOO did it come from?
* My BAR project depends on <tt>foo.jar</tt> from the FOO project.
*
  * Which build of <tt>foo.jar</tt> is used in BAR #51?
  * Which build of BAR contains my bug fix to <tt>foo.jar</tt> #32?

If someone picks wording that they like which doesn't result in the above rendering, we can use it. The change I made was the simplest thing that fixed the rendering issue while punting on the semantics.

Offhand, I have a hard time justifying a multi-level bulleted list for this. What's special about the first question bullet that makes it distinct from the other two question bullets? If the three use cases are independent and equally likely, then why give one use case doubly preferential treatment?

It sounds like the fewest character changes is swapping `.` for `:` and moving a `</li>`. But, I'm unconvinced that it's the right choice.

How does the following sound?
* I want to know where a specific `foo.jar` came from.
* I want to know which build of `foo.jar` is used in BAR version 51.
* I want to know which build of BAR has the fix in `foo.jar` version 32.

Reply:
pushed @jsoref's suggested wording